### PR TITLE
[hue] Fix scene channel updates (API v2)

### LINKED ISSUE #16000: [hue] Scene channels sometimes update wrongly due to SSE events in wrong order.
When you change a scene from the Hue App, the bridge sends two SSE events..

1. a resource for the previously active (now deactivated) scene
2. a resource for the newly activated scene

Sometimes these events are sent separately in two SSE event packages. And sometimes they are sent in a single SSE event package. In the latter case, sometimes the deactivated scene precedes the activated one. And sometimes vice versa (see JSON capture below). In this latter case the scene channel in OH will update to the wrong status. The channel is first updated to the newly activated scene id, but immediately it is updated again to UNDEF due to the second SSE event for the the deactivated scene.

```
[
    {
        "creationtime": "2023-12-03T14:47:19Z",
        "data": [
            {
                "id": "2f474323-0c6b-4361-a8c7-b52da902cd7b",
                "id_v1": "/scenes/BK-UE-77VNFHKRi",
                "status": {
                    "active": "static"
                },
                "type": "scene"
            },
            {
                "id": "9c633a2d-f7bf-4bba-b5ee-a4d69ce6e050",
                "id_v1": "/scenes/rULmu6mjOgC4U7I",
                "status": {
                    "active": "inactive"
                },
                "type": "scene"
            }
        ],
        "id": "4595793a-fd44-4a16-a406-6a274951f15e",
        "type": "update"
    }
]
```

### PR DESCRIPTION
**Pull request by jlaur**
This is a proposal for replacing the approach in #16001.

When receiving this payload:
```json
[
	{
		"creationtime": "2023-12-07T21:39:38Z",
		"data": [
			{
				"id": "00000000-0000-0000-0000-000000000001",
				"id_v1": "/scenes/0000000000000001",
				"status": {
					"active": "static"
				},
				"type": "scene"
			},
			{
				"id": "00000000-0000-0000-0000-000000000002",
				"id_v1": "/scenes/0000000000000002",
				"status": {
					"active": "inactive"
				},
				"type": "scene"
			}
		],
		"id": "00000000-0000-0000-0000-000000000000",
		"type": "update"
	}
]
```

The correct state change is triggered:
```
2023-12-07 22:39:37.368 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'Kontor_Scene' changed from Klar to Slap af
```

Also tested transition from `UNDEF` -> scene and scene -> `UNDEF`.

Since there is only one supported channel for resource type **scene**, it should be safe to skip `updateChannels()` entirely for the resource:

https://github.com/openhab/openhab-addons/blob/3814f37d9a7efae9a0d6b37e5cad4e502220ce09/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java#L933-L935

Fixes #16000

### CODE REVIEW

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:642]: .. and yes, before you say so, I do admit that my prior PR was not handling smart scenes either ..

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:643]: Just to help a bit more, getSmartSceneActive() and getSceneActive() return a tri-state result..
- Optional.ofEmpty() the resource is NOT a (smart-) scene 
- Optional.of(false) the resource is a (smart-) scene and its state is inactive
- Optional.of(true) the resource is a (smart-) scene and its state is active (actually !inactive to be precise)

**Review by andrewfg — CHANGES_REQUESTED** [review]
And I assume you will add Junit tests for this?

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: > I think you need to check for `!resource.getSceneActive().orElse(true)`

Why?

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: > OK, if you really want to handle this _perfectly_ then you need to process SMART_SCENE transitions as well. A room or a zone can have either a scene active, a smart scene active, or an inactive state. Therefore you need to handle transitions between all of those states.

I think I only want to handle that if I can reproduce that, or receive payloads for verifying such issue. Using the app I found one scene which is actually a "smart scene": natural light. But it seems it will actually just transition through some "normal" scenes, so this already works. Do you know a way to create a smart scene that will appear in the payload as active/inactive?

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: > Do you know a way to create a smart scene that will appear in the payload as active/inactive?

No.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: Because your current `ifPresent()` code returns true whether the next processed scene is active or inactive, whereas you only want to skip the scene if it is specifically inactive.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: > I think I only want to handle that if I can reproduce that

Perhaps you can understand why @maniac103 thinks that I respond too strongly. If any suggestion is your idea, it is always perfect, no matter how much work is involved for the other party. But only if it is your idea. Whereas if I have an opinion you always say it is too difficult for you to implement, and always rule it out. Perhaps a little more flexibility, please.

**Comment by andrewfg** [comment]
@jlaur your onResources() method should IMHO be as follows.
I also wrote some test case code if you would like me to share it with you..

```
    public void onResources(Collection<Resource> resources) {
        boolean sceneActivated = resources.stream().anyMatch(r -> r.getSceneActive().orElse(false) || r.getSmartSceneActive().orElse(false));
        for (Resource resource : resources) {
            boolean skipUpdate = sceneActivated && (!resource.getSceneActive().orElse(true) || !resource.getSmartSceneActive().orElse(true));
            onResource(resource, !skipUpdate);
        }
    }
```

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: In response to:
> If any suggestion is your idea, it is always perfect, no matter how much work is involved for the other party. But only if it is your idea. Whereas if I have an opinion you always say it is too difficult for you to implement, and always rule it out. Perhaps a little more flexibility, please.

and: https://github.com/openhab/openhab-addons/pull/16001#issuecomment-1847574483

Sincerely, thanks for giving me this feedback, which will allow me to reflect and self-improve. My communication skills are definitely lacking.

This is no excuse, but just to give you some explanation. Besides the mandatory review feedback about openHAB compliance, conventions, guidelines etc. I try to give individualized reviews based on multiple factors. As an example, when reviewing a high quality contribution, I might suggest improvements I probably would not have suggested otherwise. Contributor history is also a parameter - the better I know a contributor, the better I can help by knowing how much details I should go into. Some appreciate even small comments that will give a slightly better end result, others prefer a faster merge and can get frustrated by too many comments.

I have gotten the impression that you are very detail-oriented, for example based on your review of my contributions and our discussions in general. I might therefore have made some false conclusions about your preferences. I'm sorry for that, and I'll try to recalibrate.

That said, I will also give you some personal feedback. For a long time I have felt hostility from you, and at times just plain rudeness. I'm sensitive to that, so it takes energy and motivation from me. If you have felt ignored, that might have been a consequence of me simply needing to take pauses from that, i.e. reduce interactions or "throttle down" if you will. I have also already tried to put much more consideration into how I phrase things to you, to not upset you. Sometimes it slips, especially if you have really pissed me off, to be frank. It should not, but I am human. Sorry.

Going forward, besides the recalibration I will make, on short term I can try my best to stay away as contributor and reviewer of the bindings for which you are code maintainer. That should reduce tensions, and I also need that for my mental health. I can't promise for how long, since I also enjoy working on improving these bindings. After all, this was my original motivation for becoming a contributor. So it will be a balance act.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: That is intentional. There can't be more than one active scene (right?), so after reaching the first one, it doesn't make sense to process anything else.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: @jlaur many thanks for this apology; it seems that we may both be 'too detail oriented', and/or have mutually 'p**d each other off'; I therefore offer you also my apology (albeit not as eloquent as above).

My question now is how to proceed? Shall we continue with this PR? Or shall I adapt mine? I think the solution will ultimately be a combination of your code and mine. I have offered some code code below, so either you can take it over; or alternatively I can take over your code into my PR?

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: Please see my proposed code below.

**Comment by jlaur** [comment]
> your onResources() method should IMHO be as follows.

How would you like to proceed?

You challenged/invited me to propose this alternative solution I have provided here. I can go along and add smart scene support to this PR assuming that smart scenes can be inactive exactly the same way as scenes, and that scenes and smart scenes are mutually exclusive. (It just bothers me that I can't test smart scene functionality, because I can't seem to create one) - and you can review.

Or you can take over, if you think I'm too far from getting it right, and you can leave it for another maintainer if you prefer?

Regarding unit test, I don't think I'll find the time for that since there is currently no coverage for the thing handler at all (I think), so it would need to be created from scratch. I don't quite understand the test code you have provided - it uses `println()` and duplicates the code from the thing handler?

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java]: > There can't be more than one active scene (right?)

There might be two scenes activated within one SSE package.

> so after reaching the first one

We are supposed to be ignoring any IN-active messages in an SSE package, that also contains one (or more) active messages. So the suppression of the channel updates should only apply to the specifically IN-active messages.

In addition to the above, your code steps through the resources, so you may not detect a scene active until after your have received an inactive one; and thus generating the unwanted transient inactive channel state.

**Comment by andrewfg** [comment]
> I don't think I'll find the time for that since there is currently no coverage for the thing handler at all
> I don't quite understand the test code you have provided
> Or you can take over

Yes the code I posted is not Junit capable. However I do have an idea how to create a Junit test around that. Therefore let me update my prior PR as proposed. As mentioned, I will take over your top level code from this PR, so it will be a combined effort.

**Comment by andrewfg** [comment]
@jlaur I realised just now that the approach you propose in this PR is still wrong. Its aim is that if an SSE packet contains both a scene activate and a scene deactivate resource then the latter should be ignored. However this would only work if both scene resources belong to the same room or zone. By contrast if the two resources belong to different rooms or zones, then the deactivate resource should not be ignored. The more I think about this, the more I realise that this is a massive challenge, and the more I am convinced that the ‘perfect solution’ can’t make it into v4.1. My imperfect PR #16001 whilst not solving your basic requirements, would nevertheless at least allow the UI to show the correct status. So my suggestion as @maniac103 says would be to merge that one for v4.1 and get back to the perfect solution later. => WDYT?

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (andrewfg) [inline bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: Ah yes. That is certainly a step forward. And it is not so 'massive' as I had feared in the darkness of last night. I still have a few comments though..

- On line 641 and 642 you are missing a pair of brackets `ww && xx && (yy || zz)` 
- On line 643 using a forEach lambda might improve the performance a bit. ??
- On line 645 your code is still wrong; you only want to skip channel updates for deactivation events, yet your current code skips channel updates for both activation and deactivation events.
 
I said earlier that I had some thoughts about adding a Junit test. I was thinking of using static methods with an interface in the Clip2ThingHandler so it could call back a mocked inner onResource() method when running in a Junit context. But this is after all not so easy either. So there is still no Junit test that would allow you verify my assertion concerning line 645 above. As you know, I wrote a rough test suite that uses System.out.println() statements in order to verify `onResources()` and I have modified this to test both your and my current versions of onResources(). And you could choose to accept it from me that my version does pass those tests, whereas yours does not.

```
    public void onResources(Collection<Resource> resources) {
        boolean sceneActivated = resources.stream().anyMatch(r -> sceneContributorsCache.containsKey(r.getId())
                && (r.getSceneActive().orElse(false) || r.getSmartSceneActive().orElse(false)));
        resources.forEach(r -> {
            boolean skipUpdate = sceneActivated && sceneContributorsCache.containsKey(r.getId())
                    && (!r.getSceneActive().orElse(true) || !r.getSmartSceneActive().orElse(true));
            onResource(r, !skipUpdate);
        });
    }
```

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: > Ah yes. That is certainly a step forward. And it is not so 'massive' as I had feared in the darkness of last night. I still have a few comments though..

Yeah, I figured we just missed one condition, i.e. to check if the resource having scene fields belongs to the Thing. And the cache can answer this question.

> * On line 641 and 642 you are missing a pair of brackets `ww && xx && (yy || zz)`

It was unfinished, so I did one more push. 🙂 Can you check again? I tested it manually and the current version works.

> * On line 643 using a forEach lambda might improve the performance a bit. ??

Actually it's the other way around.

> * On line 645 your code is still wrong; you only want to skip channel updates for deactivation events, yet your current code skips channel updates for both activation and deactivation events.

Yes, I also realized that right after pushing, and marked as draft. This was a natural consequence of setting `sceneActivated` before the loop, so it had to be refactored. It should be okay in latest version.

> I said earlier that I had some thoughts about adding a Junit test. I was thinking of using static methods with an interface in the Clip2ThingHandler so it could call back a mocked inner onResource() method when running in a Junit context. But this is after all not so easy either. So there is still no Junit test that would allow you verify my assertion concerning line 645 above. As you know, I wrote a rough test suite that uses System.out.println() statements in order to verify `onResources()` and I have modified this to test both your and my current versions of onResources(). And you could choose to accept it from me that my version does pass those tests, whereas yours does not.

I will have a look later today. If this means the tests would have to be run and checked manually by a developer rather than by CI, I'm not sure how much value it provides? Since there is no test coverage of the thing handler now anyway, we could go ahead without for 4.1, and add proper test coverage later. There are many bindings to take inspiration from with regards to proper mocking and testing of handlers.

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: >> It was unfinished, so I did one more push. 🙂 Can you check again?

Unfortunately it is still not right. I know that you have a thing about removing excess brackets in complex logic operations. And frankly I am not 100% sure about the precedence order of `||` vs `&&`. But in this case I think it really helps to use more brackets than fewer in order to make the logic explicit rather than implicit..

```
// yours
.anyMatch(r -> sceneContributorsCache.containsKey(r.getId()) && r.getSceneActive().orElse(false) || r.getSmartSceneActive().orElse(false));
// mine
.anyMatch(r -> sceneContributorsCache.containsKey(r.getId()) && (r.getSceneActive().orElse(false) || r.getSmartSceneActive().orElse(false)));
```

``` 
// yours
boolean skipUpdate = sceneActivated
                    && !(resource.getSceneActive().orElse(false) || resource.getSmartSceneActive().orElse(false));
// mine
boolean skipUpdate = sceneActivated && sceneContributorsCache.containsKey(r.getId())
                    && (!r.getSceneActive().orElse(true) || !r.getSmartSceneActive().orElse(true));
```

Note: concerning the latter above: I am not sure if your outer inversion with orElse(false) does actually evaluate identical to my inner inversion with orElse(true) .. maybe / maybe not. But I can tell you that my version has at least been tested manually.

Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: That was a clear bug, thanks. Fixed.

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: Apropos Junit testing..
1. I have no objection to postponing the introduction of a proper Junit test
2. Nevertheless once you are ready I will still do a final manual test before I approve the PR

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: > once you are ready I will still do a final manual test before I approve the PR

@jlaur let me know when you are ready (the current version fails my tests)

Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: How does it fail?

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: > How does it fail?

See below.
The problem is the logic in your line 644 .. where I have already posted my suggestion above.

```
testTransitionA
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:scene => skipped
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:scene => processed
Resource id:2, type:light => processed
Resource id:3, type:scene => skipped
Resource id:4, type:light => processed

testTransitionB
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:scene => skipped
Resource id:2, type:light => skipped
Resource id:3, type:scene => processed
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:scene => skipped
Resource id:2, type:light => processed
Resource id:3, type:scene => processed
Resource id:4, type:light => processed

testTransitionC
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:smart_scene => skipped
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:scene => processed
Resource id:2, type:light => processed
Resource id:3, type:smart_scene => skipped
Resource id:4, type:light => processed

testTransitionD
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:scene => skipped
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => processed
Resource id:3, type:scene => skipped
Resource id:4, type:light => processed

testTransitionE
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:smart_scene => skipped
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => processed
Resource id:3, type:smart_scene => skipped
Resource id:4, type:light => processed

testTransitionF
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:scene => processed
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:scene => processed
Resource id:2, type:light => processed
Resource id:3, type:scene => processed
Resource id:4, type:light => processed

testTransitionG
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => skipped
Resource id:3, type:smart_scene => processed
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:smart_scene => processed
Resource id:2, type:light => processed
Resource id:3, type:smart_scene => processed
Resource id:4, type:light => processed

testTransitionH
jlaur code
Resource id:0, type:light => processed
Resource id:1x, type:scene => processed
Resource id:2, type:light => processed
Resource id:3, type:scene => processed
Resource id:4, type:light => processed
andrewfg code
Resource id:0, type:light => processed
Resource id:1x, type:scene => processed
Resource id:2, type:light => processed
Resource id:3, type:scene => processed
Resource id:4, type:light => processed

testTransitionI
jlaur code
Resource id:0, type:light => skipped
Resource id:1, type:scene => processed
Resource id:2, type:light => skipped
Resource id:3x, type:scene => skipped
Resource id:4, type:light => skipped
andrewfg code
Resource id:0, type:light => processed
Resource id:1, type:scene => processed
Resource id:2, type:light => processed
Resource id:3x, type:scene => processed
Resource id:4, type:light => processed
```

Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: Sorry for being slow, you are right. I have given it another go inverting some things to (hopefully) make it simpler/easier to read without double negations. Does it parse your tests, and how do you like this version?

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: @jlaur thanks for the update; I will run the tests again and give (hopefully) final approval tomorrow.

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: @jlaur I checked your code. I think the logic inversion is a good improvement, and the left to right 'or' will probably improve performance. However I think there is still an issue concerning the two 'orElse(false)' statements, which with the logic inversion should be changed to 'orElse(true)'. This handles the cases where there may be an empty scene resource that contains neither an activate nor a deactivate element (but perhaps something else). I think those cases should be allowed through the gate to update the channels too. The updateChannels method currently has no use for such resources, but I think we should not close the gate via this code here, but rather let updateChannels make that decision.

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: ^
HOWEVER I realise that simply changing simply to 'orElse(true)' is also wrong, since the two combined would always cause the resource to pass through to updateChannels. Aargh!

Comment (jlaur) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: I think if there would be "something else" related to scenes (other than currently active scene) that we would need to handle, this logic would need to be adapted/rewritten anyway, since it's based on the assumption that we are only updating that channel. So I'm wondering if there is actually anything reasonable to do at this moment?

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: Yeah. The 'something else' case would be when it is a scene resource and both getSceneActive and getSmartSceneActive are empty.

Comment (andrewfg) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java:639]: So let us leave the code as it is for the moment.

**Review by andrewfg — APPROVED** [review]
LGTM (finally)

**Comment by gcatellani** [comment]
Thanks for all the effort and for overcoming your differences to provide us with a quick fix and a more thorough one as well as the binding in general! It‘s very much appreciated.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**
